oar/core/shipment.py

```diff
@@ -79,14 +84,64 @@ def get_all_files(self, file_extension: str = None) -> List[str]:
                 return [f for f in all_files if f.lower().endswith(f'.{file_extension.lower()}')]
             return all_files
         except Exception as e:
-            raise Exception(f"Failed to get changed files: {str(e)}")
+            raise GitLabMergeRequestException(f"Failed to get changed files: {str(e)}")
 
     def add_comment(self, comment: str) -> None:
         """Add comment to the merge request"""
         try:
             self.mr.notes.create({'body': comment})
         except Exception as e:
-            raise Exception(f"Failed to add comment: {e}")
+            raise GitLabMergeRequestException(f"Failed to add comment: {e}")
+
+
+class GitLabServer:
+    """Class for performing GitLab server-level operations"""
+    
+    def __init__(self, gitlab_url: str, private_token: str = None):
+        """Initialize with GitLab connection details
+        
+        Args:
+            gitlab_url: URL of GitLab instance (hostname only)
+            private_token: Optional token, falls back to GITLAB_TOKEN env var
+        """
+        self.gitlab_url = gitlab_url
+        self.private_token = private_token or os.getenv('GITLAB_TOKEN')
+        
+        if not self.private_token:
+            raise ValueError("No GitLab token provided and GITLAB_TOKEN env var not set")
+            
+        self.gl = gitlab.Gitlab(gitlab_url, private_token=self.private_token)
+        
+    def get_username_by_email(self, email: str) -> Optional[str]:
+        """Query GitLab username by email address
+        
+        Args:
+            email: email address to search for
+            
+        Returns:
+            first matching GitLab username if found, None otherwise
+            
+        Raises:
+            ValueError: if email is empty, not a string, or invalid format
+            Exception: if there are errors accessing GitLab API
+        """
+        if not email or not isinstance(email, str):
+            raise ValueError("Email must be a non-empty string")
+            
+        if not validate_email(email):
+            raise ValueError("Email must be in valid format (e.g. user@example.com)")
+            
+        if not email.strip():
```

Comment:
Isn't this already checked by the email validation function? I think that the regex already checked the correct format.

Reply:
removed unused checkpoints